src/data_processing/netflix_to_wikidata.py

```diff
@@ -0,0 +1,110 @@
+import os
+from dotenv import load_dotenv
+import requests
+import pandas as pd
+
+load_dotenv()
+
+base_dir = os.getenv('BASE_DIR')
+user_agent = os.getenv('USER_AGENT')
+
+netflix_data_path = os.path.join(base_dir, 'netflix_movies.txt')
+output_csv_path = os.path.join(base_dir, 'netflix_to_wikidata.csv')
+
+netflix_data = pd.read_csv(netflix_data_path, delimiter=',', names=['NetflixId', 'Year', 'Title'],  encoding='ISO-8859-1', on_bad_lines='skip')
+
+def construct_query(title, year):
+    query = '''
+    SELECT * WHERE {
+        SERVICE wikibase:mwapi {
+            bd:serviceParam wikibase:api "EntitySearch" ;
+                            wikibase:endpoint "www.wikidata.org" ;
+                            mwapi:search "%s" ;
+                            wikibase:limit 1 ;
+                            mwapi:language "en" .
+            ?item wikibase:apiOutputItem mwapi:item .
+        }
+
+        ?item wdt:P31/wdt:P279* ?type .
+        FILTER (?type IN (
+            wd:Q11424,    # Film
+            wd:Q5398426,  # Television Series
+            wd:Q1259759,  # Television Miniseries
+            wd:Q202866,   # Animated Film
+            wd:Q24862,    # Documentary Film
+            wd:Q506240,   # Short Film
+            wd:Q204370,   # Television Film
+            wd:Q471839,   # Animated Television Series
+            wd:Q21191270, # Web Series
+            wd:Q7725310,  # Reality Television Series
+            wd:Q579956,   # Anthology Series
+            wd:Q15416,    # Television Program
+            wd:Q28018927  # Television Special
+        )).
+
+        OPTIONAL {
+            ?item wdt:P577 ?releaseDate .
+            FILTER (YEAR(?releaseDate) >= %d && YEAR(?releaseDate) <= %d) .
+        }
+
+        OPTIONAL {
+            ?item wdt:P136 ?genre .
+            ?genre rdfs:label ?genreLabel .
+            FILTER (lang(?genreLabel) = "en") .
+        }
+
+        ?item rdfs:label ?itemLabel .
+        FILTER (lang(?itemLabel) = "en") .
+
+        SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
+    }
+    LIMIT 1
+    ''' % (title, year - 2, year + 2)
```

Comment:
when you change it to `%(search_term)`, you will have to fill it with a dictionary instead of a tuple.